packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -26,226 +24,93 @@ const getClassNames = classNamesFunction<ISwatchColorPickerStyleProps, ISwatchCo
-    } = this.props;
-
-    const classNames = getClassNames(styles!, {
-      theme: this.props.theme!,
-      className,
-      cellMargin,
-    });
-
-    if (colorCells.length < 1 || columnCount < 1) {
-      return null;
-    }
-    return (
-      <Grid
-        {...this.props}
-        id={this._id}
-        items={this._getItemsWithIndex(colorCells)}
-        columnCount={columnCount}
-        onRenderItem={this._renderOption}
-        ariaPosInSet={ariaPosInSet}
-        ariaSetSize={ariaSetSize}
-        shouldFocusCircularNavigate={shouldFocusCircularNavigate}
-        doNotContainWithinFocusZone={doNotContainWithinFocusZone}
-        onBlur={this._onSwatchColorPickerBlur}
-        theme={this.props.theme!}
-        styles={{
-          root: classNames.root,
-          tableCell: classNames.tableCell,
-          focusedContainer: classNames.focusedContainer,
-        }}
-      />
-    );
-  }
-
   /**
    * When the whole swatchColorPicker is blurred,
    * make sure to clear the pending focused stated
    */
-  private _onSwatchColorPickerBlur = (): void => {
-    if (this.props.onCellFocused) {
-      this._cellFocused = false;
-      this.props.onCellFocused();
+  const onSwatchColorPickerBlur = (): void => {
+    if (props.onCellFocused) {
+      cellFocused = false;
+      props.onCellFocused();
     }
   };
 
-  /**
-   * Get the selected item's index
-   * @param items - The items to search
-   * @param selectedId - The selected item's id to find
-   * @returns - The index of the selected item's id, -1 if there was no match
-   */
-  private _getSelectedIndex(items: IColorCellProps[], selectedId: string): number | undefined {
-    const selectedIndex = findIndex(items, item => item.id === selectedId);
-    return selectedIndex >= 0 ? selectedIndex : undefined;
-  }
-
-  /**
-   * Render a color cell
-   * @param item - The item to render
-   * @returns - Element representing the item
-   */
-  private _renderOption = (item: IColorCellProps): JSX.Element => {
-    const props = this.props;
-    const id = this._id;
-
-    return (
-      <ColorPickerGridCell
-        item={item}
-        idPrefix={id}
-        color={item.color}
-        styles={props.getColorGridCellStyles}
-        disabled={props.disabled}
-        onClick={this._onCellClick}
-        onHover={this._onGridCellHovered}
-        onFocus={this._onGridCellFocused}
-        selected={this.state.selectedIndex !== undefined && this.state.selectedIndex === item.index}
-        circle={props.cellShape === 'circle'}
-        label={item.label}
-        onMouseEnter={this._onMouseEnter}
-        onMouseMove={this._onMouseMove}
-        onMouseLeave={this._onMouseLeave}
-        onWheel={this._onWheel}
-        onKeyDown={this._onKeyDown}
-        height={props.cellHeight}
-        width={props.cellWidth}
-        borderWidth={props.cellBorderWidth}
-      />
-    );
+  const getSelectedIndex = (items: IColorCellProps[], selectedId: string): number | undefined => {
```

Comment:
This can go outside the component body (and bring back at least the first line of the doc comment)